spec/android_download_file_by_version_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'spec_helper'
+
+describe Fastlane::Actions::AndroidDownloadFileByVersionAction do
+  describe 'run' do
+    let(:test_repo) { 'repo-test/project-test' }
+    let(:test_tag) { 'release/10.0' }
+    let(:test_version) { '10.0' }
+    let(:test_file) { 'test-file.xml' }
+    let(:test_folder) { 'test-folder' }
+    let(:test_token) { 'GITHUB_TOKEN' }
+    let(:github_helper) { class_double(Fastlane::Helper::GithubHelper) }
+
+    before do
+      allow(github_helper).to receive(:download_file_from_tag) # .and_return(nil)
+      allow(Fastlane::Helper::GithubHelper).to receive(:new).and_return(github_helper)
+      allow(Fastlane::Helper::Android::VersionHelper).to receive(:get_library_version_from_gradle_config).and_return(test_version)
+    end
+
+    it 'includes the github_token' do
+      expect(Fastlane::Helper::GithubHelper).to receive(:new).with(github_token: test_token)
+      described_class.run(mock_params)
+    end
+
+    it 'calls the download_file_from_tag' do
+      expect(github_helper).to receive(:download_file_from_tag).with(
+        repository: test_repo,
+        tag: test_tag,
+        file_path: test_file,
+        download_folder: test_folder
+      )
+
+      described_class.run(mock_params)
+    end
+
+    def mock_params
+      {
+        repository: test_repo,
+        github_release_prefix: 'release/',
+        file_path: test_file,
+        download_folder: test_folder,
+        github_token: test_token
+      }
+    end
+  end
+
+  describe 'available_options' do
+    it 'has the correct length' do
+      expect(described_class.available_options.length).to be(7)
+    end
+
+    it 'has the ConfigItem library_name' do
+      expect(described_class.available_options).to include(an_object_having_attributes(key: :library_name))
+    end
+
+    it 'has the ConfigItem import_key' do
+      expect(described_class.available_options).to include(an_object_having_attributes(key: :import_key))
+    end
+
+    it 'has the ConfigItem repository' do
+      expect(described_class.available_options).to include(an_object_having_attributes(key: :repository))
+    end
+
+    it 'has the ConfigItem file_path' do
+      expect(described_class.available_options).to include(an_object_having_attributes(key: :file_path))
+    end
+
+    it 'has the ConfigItem download_folder' do
+      expect(described_class.available_options).to include(an_object_having_attributes(key: :download_folder))
+    end
+
+    it 'has the ConfigItem github_release_prefix' do
+      expect(described_class.available_options).to include(an_object_having_attributes(key: :github_release_prefix))
+    end
+
+    it 'has the ConfigItem github_token' do
+      expect(described_class.available_options).to include(an_object_having_attributes(key: :github_token))
+    end
+  end
```

Comment:
Mmmh I like the addition of more unit tests, but tbh I'm not convinced those are particularly useful ones.
They are kind of self-testing, and as soon as we might want to add a new `ConfigItem` (potentially optional and with default value, which would make it non-API-breaking) to the action in the future to add some flexibility, we'll have to update the unit tests to reflect… exactly the same. It's a bit like testing `expect(a).to eq(1)` right after `a=1` all over again 😅 . It's not that important, for the **behavior** of the action under test to be correct, that the number of `ConfigItems` of the action is checked to be exactly `7`, for example.

I think it's more useful when the unit test ensures the result of the action, i.e. the output at the end of the pipeline, based on the input parameters, to test the behavior of the action itself, rather than just testing trivialities at input. 

Like, it's more useful to test that when you pass a specific value to the `github_token` parameter when calling `described_action.run`, that token is the one being used in the REST API call / `Octokit::Client` that is made by the action (instead of being deduced by the helper via the value of an `ENV` var, like before) — which is what you did on [your test on lines 19-22](https://github.com/wordpress-mobile/release-toolkit/pull/423/files#diff-9ab3e8e35d6b48ba7b7484e5fc91ba5f4e78334c4c7e96074f22df068bcf5fc5R19-R22), rather than just to test that the action has a `ConfigItem` with a `github_token` key like you do here.